index.js

```diff
@@ -232,6 +232,31 @@ Renderer.prototype.updatePointLightShadowMap = function (light, geometries) {
   })
 }
 
+Renderer.prototype.parseShader = function (string, options) {
```

Comment:
@vorg the regex has some constraints (don't mess up with spaces, need to replace the NUM_LIGHTS as well) but it is working fine. It should provide a tiny perf boost by not evaluating a for loop.
I opted to parse the shader only on program build, in order to call it only once and not every frame